xdsl/irdl.py

```diff
@@ -257,7 +257,8 @@ def irdl_to_attr_constraint(
         generic_args = ()
 
         # Get the Generic parent class to get the TypeVar parameters
-        for parent in origin.__orig_bases__:  # type: ignore
+        bases = cast(Sequence[Any], origin.__orig_bases__)  # type: ignore
```

Comment:
Just a sanity check, do we still need the type: ignore here?
Or is it because it doesn't know about `__orig_bases__`?

Reply:
Unfortunately yes, although weirdly only for `pyright`, not Pylance in vscode. There are two errors on this line, the fact that `origin` is unknown, and the fact that `origin.__orig_bases__` is unknown. The `type: ignore` here is for the first one, and the cast is for the second